.storybook/main.ts

```diff
@@ -8,6 +9,22 @@ const config: StorybookConfig = {
   docs: { autodocs: 'tag' },
   staticDirs: ['../public'],
   core: { disableTelemetry: true },
+  webpackFinal: async (config, { configType }) => {
+    config.resolve!.modules = [path.resolve(__dirname, '..'), 'node_modules'];
+
+    config.resolve!.alias = {
```

Comment:
Custom Webpack Configuration for storybook
https://storybook.js.org/docs/react/builders/webpack#import-a-custom-webpack-configuration

Reply:
Sure, but that's not my question. Why are you doing it like this? (Like why are you changing `config.resolve.modules?`

```tsx
webpackFinal: (config) => {
  config.resolve.alias = {
    ...config.resolve.alias,
      "@/": path.resolve(__dirname, '..'),
  };

  return config;
},
```

Pretty much I want to understand why you're doing it this way.

Reply:
Because storybook doesn't support Next.js absolute paths out of the box 
https://github.com/storybookjs/storybook/issues/11639

Reply:
Sorry, let me rephrase my question. First, yes I got that you made this custom config because Storybook does not support the Next.js aliases.

My question was about why the config.modules, but I suppose that after looking at the linked issue, I think I get what is being done there. Still I don’t understand why you added manual aliases for each folder. On the linked issue they’ve done with a simple @ alias 🤔

Reply:
Like this:

```tsx

webpackFinal: async (config, { configType }) => {
    config.resolve.modules = [path.resolve(__dirname, '..'), 'node_modules'];
    config.resolve.alias = {
      ...config.resolve.alias,
      '@': path.resolve(__dirname, '../'),
    };
    return config;
  },

```

Reply:
Also that resolve could become a simple constant on the beginning of the file for reusability sake

Reply:
> Sorry, let me rephrase my question. First, yes I got that you made this custom config because Storybook does not support the Next.js aliases.
> 
> My question was about why the config.modules, but I suppose that after looking at the linked issue, I think I get what is being done there. Still I don’t understand why you added manual aliases for each folder. On the linked issue they’ve done with a simple @ alias thinking

I tried that but it didn't pick up so I figured these values should be hard coded.

Reply:
This definitely requires more investigation. Hard-coding this is sadly a no-go :(

Reply:
I agree with that

Reply:
> Like this:
> 
> ```tsx
> webpackFinal: async config => {
>     config.resolve.modules = [resolve(__dirname, '..'), 'node_modules'];
>     config.resolve.alias = {
>       ...config.resolve.alias,
>       '@': resolve(__dirname, '../'),
>     };
>     return config;
>   },
> ```

FWIW, this worked fine for me locally. I was also able to run storybook build and serve it.

Reply:
I will look into this once more if its running for you then I should be doing something wrong.

Reply:
@mikeesto It is failing for me in local as well

![image](https://github.com/nodejs/nodejs.org/assets/52033975/726c65f7-23ad-4cb3-9045-1c60637ad6c9)

Reply:
For me, it looks like a cache issue. The approach above also works for me.

You can test that it works in CI by committing the example. Or @mikeesto can take-over the PR, at least with this change. Maybe he tweaked it a little bit.

Reply:
got it I was confused with the `/` turns out `/` was causing the issue with me and I thought that `/` was also required for directory mapping

Reply:
Yeah the `@` is not required at all 🙈 actually should not even be there! What we're aliasing is `@` and not `@/` because `@/` is already two different directories.

Glad to hear you solved it!